### ISSUE TITLE
Consistent Notifications on Health Check Failures

### ISSUE THREAD

**Issue by DrSeussFreak** [Sonarr/Sonarr#3514]
**Describe the bug**
I continually receive pushover notifications about health check failures.  I have specifically unchecked the option to receive notifications about health check failures.

**Screenshots**
![Screenshot_20200113-121427](https://user-images.githubusercontent.com/175500/72280728-9a01dc80-35fe-11ea-9f94-092ec06f8655.png)

![Screen Shot 2020-01-13 at 12 14 50 PM](https://user-images.githubusercontent.com/175500/72280697-8c4c5700-35fe-11ea-8874-4def6bb67b6f.png)

**System Information**
 - Sonarr Version: 3.0.3.688
 - Operating System: Ubuntu 19.10
 - .net Framework (Windows) or mono (macOS/Linux) Version: Mono: 6.8.0.96

**Additional context**
This has been going on for a few months, it comes and goes.  my indexers are working fine, sonarr is grabbing files without issue.

**Comment by markus101** [comment]
> my indexers are working fine, sonarr is grabbing files without issue.

If they're all being disabled then either there is an issue with them or the network.

> I have specifically unchecked the option to receive notifications about health check failures.

Is the issue you're getting health check notifications even though the option is disabled or that you're getting them about your indexer(s) period?

If it's the former, we'll need debug logs. If it's the latter then this is not the place to troubleshoot why it's happening, but you'd need to look at debug logs and figure out why they're failing.

If you had a single indexer that was unavailable when Sonarr went to check the RSS feed from it, then it'd be marked as failed and you'd get a notification, even though it may be available later and you wouldn't miss anything.

**Comment by DrSeussFreak** [comment]
The issue is that I am getting notified at all about the health check status.  I disabled the check mark to avoid this, but no matter the status of it being checked, I receive these notifications.

I have multiple indexers listed, majority are via nzbhydra2, but I have some torrent sites as well.  if I login immediately after receiving this alert, nothing is marked as wrong.  I have seen the errors before, they show up in orangish/yellow, there are no errors at that time, and if I go to indexers and test any (or all), they work without issue.

I believe I am getting false positives, and honestly, I have monitoring on my environment, so if nzbhydra2 went down, I will know.  I just want to stop receiving health check failures, like I specified.

**Comment by markus101** [comment]
With Pushover not set to send On Health notifications I am receiving no notifications from Sonarr when something fails the health check.

Trace logs will show the following when a health failure leads to a Pushover notification being sent:

```
20-1-13 17:09:57.1|Trace|EventAggregator|Publishing HealthCheckFailedEvent
20-1-13 17:09:57.1|Trace|EventAggregator|HealthCheckFailedEvent -> NotificationService
20-1-13 17:09:57.9|Debug|RestSharpExtensions|Validating Responses from [POST] [https://api.pushover.net/1/messages.json] status: [OK]
20-1-13 17:09:57.9|Trace|EventAggregator|HealthCheckFailedEvent <- NotificationService
```

And this when On Health is disabled for Pushover:

```
20-1-13 17:11:13.1|Trace|EventAggregator|Publishing HealthCheckFailedEvent
20-1-13 17:11:13.2|Trace|EventAggregator|HealthCheckFailedEvent -> NotificationService
20-1-13 17:11:13.2|Trace|EventAggregator|HealthCheckFailedEvent <- NotificationService
```

My guess is something else (Radarr or Lidarr perhaps) was setup to send health check notifications through the Sonarr app (it's based on API key) and they sending it.

**Comment by DrSeussFreak** [comment]
I checked radarr, but there is no option for it at all. It still uses the old framework, it never upgraded to this interface.

I have lidarr, and it was set to send health notifications, but when i hit test it was clearly marked from lidarr, and the alerts I've been receiving were marked sonarr. Regardless, I changed the settings on lidarr, I'll report back after a day or two of i receive more alerts or not.
